simpleviewinc/keg-hub
# WIP: Removed unused repos

## Context
* Remove unused repos
* Move Keg-CLI to it's own repo
  * I had some issue with migrating some of the branches, so I had to make a few updates
  * I few things have been migrated into the `keg-cli` as you'll see in the tests below
* Keg-CLI has been migrated to => https://github.com/simpleviewinc/keg-cli
  * Any future bugs / issues for the cli, should now be address in that repo.
  * There should be no other PR's changes to the keg-cli, to ensure it's remove properly from this repo
    * Which is why I was running into issues migrating some of the of PRs / branches   

## Updates

* Remove `retheme/app`, `keg-regulator` and `keg-cli`
* Updated keg-cli tap links to be objects with a path
  * This will eventually be used for custom tap tasks 
  * This includes an update to your global cli config
* Updated the `KEG_IMAGE_FROM` env to be `keg-base:development` instead of `keg-base:latest`
  * This is an update to your global cli config
  * Technically you don't need to make this change if your keg-base image is tagged with `latest`
  * But unless you know what you're doing, I would recommend doing this update

## Testing

**Test 1**
* Pull this PR => `keg && keg pr 123`
* This will remove the keg-cli, so we need to replace it
* Run this command => `cd ~/keg-hub/repos && git clone https://github.com/simpleviewinc/keg-cli.git`
  * This will put the keg-cli back in its old location, but under a different git repo
* Next run `cd ./keg-cli && yarn install`
  * You should now have the `keg-cli` commands back
  * Run `keg test` to make sure the `keg` executable is in your path

**Test 2**
**Important**
* There were a couple changes made, so you'll need to run a couple scripts
* Update Tap Links by running => `keg cli && node scripts/cli/updates/updateTapsPaths.js`
  * You can run the script as many times as you want, so feel free
  * If your tap links have already been updated it looks like this => http://snpy.in/jGl4Fv 
  * Your global cli config should now look similar to this => http://snpy.in/J6PVf6
    * All it's doing it converting your global config tap links into objects with a path property
* After that run `keg retheme`
  * Ensure the `retheme` tap link works
* Next run `keg cli env sync --no-merge --conflict local`
  * This should update `KEG_IMAGE_FROM` env to use keg-base develop 
  * Looks like this => http://snpy.in/TshRQv
* Now remove and rebuild the `keg-base` image with command `keg di rm keg-base && keg base build` 
  * It should tag your keg-base image with `development` automatically
  * Looks like this => http://snpy.in/TFMmK8

**Test 3**
* Next run `keg cli test`
 * Ensure all tests pass
 * **Note**, some tests expect `keg-base` image to exist, so make sure you have that image built locally
   * Run `keg base build`, if you don't already have it

**Test 4**
* First remove `retheme` docker image, if you already have it
* Next run `keg retheme start` 
  * It should build and run the link re-theme tap
  * **Note** - It might show warnings when running
    * This is because the latest re-theme requires updates to `jsutils` that are not published 
    * If you want to make sure it working, run `keg retheme sync jsutils:install:build`, to sync and install jsutils
